pm_nq/pm_nq.xml

```diff
@@ -0,0 +1,253 @@
+            pm-nq -t w \$structure_name params.yaml &&
+            find * > tree.txt &&
+            zip -r out_zip.zip *
+        #else if $task.task == 'r':
+            unzip '$task.zipped_input' &&
+            pm-nq -t r \$structure_name params.yaml &&
+            cat averages.dat
+        #end if
+    ]]></command>
+    <configfiles>
+        <configfile name="params_yaml">#if $task.task == 'w':
+method: $task.method
+mu_symbol: $task.mu_symbol
+mu_index: $task.mu_index
+grid_n: $task.grid_n
+sigma_n: $task.sigma_n
+#set $k_points_grid_formatted = $task.k_points_grid.replace('__ob__', '[').replace('__cb__', ']')
+k_points_grid: $k_points_grid_formatted
+avgprop: $task.avgprop
+calculator: $task.calculator.calculator
+write_allconf: $task.write_allconf
+displace_T: $task.displace_T
+#if $task.average_T:
+average_T: $task.average_T
+#end if
+#if $task.random_seed:
+random_seed: $task.random_seed
+#end if
+#if $task.calculator.calculator == 'dftb+':
+dftb_set: $task.calculator.dftb_set
+dftb_pbc: $task.calculator.dftb_pbc
+#end if
+phonon_source_type: $task.phonon.phonon_source_type
+#if $task.phonon.phonon_source_type == "castep":
+#set $phonon_source_file_name = $task.phonon.phonon_source_castep.name.replace(' ', '')
+#else if $task.phonon.phonon_source_type == "dftb+":
+#set $phonon_source_file_name = $task.phonon.phonon_source_dftb.name.replace(' ', '')
+#end if
+phonon_source_file: $phonon_source_file_name
+#if $task.script_file:
+#set $script_file_name = $task.script_file.name.replace(' ', '')
+script_file: $script_file_name
+#end if
+#if $task.calculator.calculator == 'castep' and $task.calculator.castep_param:
+#set $castep_param_name = $task.calculator.castep_param.name.replace(' ', '')
+castep_param: $castep_param_name
+#end if
+#end if</configfile>
+    </configfiles>
+    <inputs>
+        <param argument="structure" type="data" format="cell, cif, xyz, extxyz" label="Structure File" help="The structure to make displacements to. The original file will be preserved. Accepted file types: CELL, CIF, XYZ (standard or extended)."/>
+        <conditional name="task">
+            <param argument="task" type="select" display="radio" label="Task" help="Task to perform with pm-nq. Either generate a displaced structures, or average those structures.">
+                <option value="w">Generate</option>
+                <option value="r">Average</option>
+            </param>
+            <when value="w">
+                <conditional name="phonon">
+                    <param argument="phonon_source_type" type="select" display="radio" label="Phonon Type" help="The format of the phonon file.">
+                        <option value="castep">CASTEP</option>
+                        <option value="dftb+">DFTB+</option>
+                    </param>
+                    <when value="castep">
+                        <param name="phonon_source_castep" type="data" format="txt" label="Phonon File" help="Source file for phonon modes."/>
```

Comment:
I assume these are the phonons for the un-muonated structure? Might be good to make that explicit.

Reply:
It's actually for the muonated structure (we're displacing the muon so need its phonon modes) - I'll make that explicit